app/src/main/java/com/infomaniak/drive/views/ShareLinkContainerView.kt

```diff
@@ -45,44 +45,79 @@ class ShareLinkContainerView @JvmOverloads constructor(
     fun setup(
         file: File,
         shareLink: ShareLink? = null,
-        onTitleClicked: (shareLink: ShareLink?, currentFileId: Int) -> Unit,
-        onSettingsClicked: (shareLink: ShareLink, currentFile: File) -> Unit
+        onTitleClicked: ((shareLink: ShareLink?, currentFileId: Int) -> Unit)? = null,
+        onSettingsClicked: ((shareLink: ShareLink, currentFile: File) -> Unit)? = null,
     ) {
         currentFile = file
         this.shareLink = shareLink
         isVisible = true
         urlValue = file.shareLink ?: ""
 
-        updateUi()
+        selectUI(file.isDropBox())
 
-        titleContainer.setOnClickListener { onTitleClicked(this.shareLink, currentFile.id) }
-        // cannot be null, if null, settings will not appear
-        shareLinkSettings.setOnClickListener { onSettingsClicked(this.shareLink!!, currentFile) }
+        if (file.isDropBox()) return
 
+        titleContainer.setOnClickListener { onTitleClicked?.invoke(this.shareLink, currentFile.id) }
+        // cannot be null, if null, settings will not appear
+        shareLinkSettings.setOnClickListener { onSettingsClicked?.invoke(this.shareLink!!, currentFile) }
         shareLinkButton.setOnClickListener { this.shareLink?.url?.let(context::shareText) }
```

Comment:
I think that'll be good without the return here, 
You need make this if only it's not a dropBox